tests/test_cli_compile.py

```diff
@@ -3455,7 +3464,6 @@ def test_compile_recursive_extras_build_targets(runner, tmp_path, current_resolv
     expected = rf"""foo[footest] @ {tmp_path.as_uri()}
 small-fake-a==0.2
 small-fake-b==0.3
-wheel==0.42.0
```

Comment:
I don't think it's acceptable to both mark the test as xfail and make changes to it simultaneously.

Reply:
the test above marked with xfail is this one:
`test_preserve_via_requirements_constrained_dependencies_when_run_twice()`
whereas the test being adjusted is this one:
`test_compile_recursive_extras_build_targets()`

i can see how it may look like the same test though

Reply:
Oh, that's what I missed. In that case, this change is incorrect, as it masks the problem that #2106 is attempting to fix.